### ISSUE TITLE
Exercise thumbnail point displays

### ISSUE THREAD

**Issue by s-edwards** [web-cat/code-workout#13]
Exercise thumbnails need to include meaningful point displays even when the exercise is not associated with the current workout, and need to include meaningful points earned info if the current user has completed (or attempted) it.  At present, the logic for displaying points earned + total points in thumbnails is incomplete and needs to be redone.  It does not appropriately tie into XP when exercises are outside of assigned workouts.

**Comment by pmk1433** [comment]
Have done most of the requirements for this issue, though am not exactly clear on 'tie into XP when exercises are outside of assigned workouts.'

**Comment by s-edwards** [comment]
Three issues: first, there are two exercise thumbnails and they both need to be addressed.  I duplicated your changes to exercises/thumblg over in exercises/thumbsm.  Second, exercises might be viewed both when there is a current workout for the current user, and when there is not.  There were nil errors in your changes in the second case, so be sure to test changes under both conditions to make sure things are working correctly.  Third, an exercise thumbnail might correspond to a context where it is being viewed as part of a workout where it will earn points, or in a context where it is not (say, one simply searches for exercises to practice in the gym, without being enrolled in a course or anything).  Instead of showing points (earned toward a workout score), the exercise should show xp in this case.

Basically, exercises always have xp, and completing them always earns xp, but in the case where an exercise is also part of a workout the user is completing, and the workout has allocated a specific point value to the exercise, that exercise /also/ earns points toward the workout.

**Comment by pmk1433** [comment]
I believe I have made substantial progress towards the completion of this issue in my latest commit. Please check it out and let me know if there is anything more that needs to be done.

**Comment by pmk1433** [comment]
Can this be reviewed in PROD and closed out?

**Comment by pmk1433** [comment]
Closing this out since it has been incorporated long ago.
